internal/controller/accessrequest_controller.go

```diff
@@ -350,52 +417,66 @@ func (r *AccessRequestReconciler) grantArgoCDAccess(ctx context.Context, ar *api
 
 // addSubjectsInRole will associate the given AccessRequest subjects in the
 // specific role in the given project.
-func addSubjectsInRole(project *argocd.AppProject, ar *api.AccessRequest) bool {
-	modified := false
+func addSubjectsInRole(project *argocd.AppProject, ar *api.AccessRequest, rt *api.RoleTemplate) {
 	roleFound := false
-
+	roleName := rt.AppProjectRoleName(ar.Spec.Application.Name, ar.Spec.Application.Namespace)
 	for idx, role := range project.Spec.Roles {
-		if role.Name == ar.Spec.TargetRoleName {
+		if role.Name == roleName {
 			roleFound = true
 			for _, subject := range ar.Spec.Subjects {
 				hasAccess := false
-				for _, groupClaim := range role.Groups {
-					if groupClaim == subject.Username {
+				for _, group := range role.Groups {
+					if group == subject.Username {
 						hasAccess = true
 						break
 					}
 				}
 				if !hasAccess {
-					modified = true
 					project.Spec.Roles[idx].Groups = append(project.Spec.Roles[idx].Groups, subject.Username)
 				}
 			}
 		}
 	}
 	if !roleFound {
-		addRoleInProject(project, ar)
-		modified = true
+		addRoleInProject(project, ar, rt)
 	}
-	return modified
 }
 
 // addRoleInProject will initialize the role owned by the ephemeral-access
 // controller and associate it in the given project.
-func addRoleInProject(project *argocd.AppProject, ar *api.AccessRequest) {
+func addRoleInProject(project *argocd.AppProject, ar *api.AccessRequest, rt *api.RoleTemplate) {
 	groups := []string{}
 	for _, subject := range ar.Spec.Subjects {
 		groups = append(groups, subject.Username)
 	}
 	role := argocd.ProjectRole{
-		Name:        ar.Spec.TargetRoleName,
-		Description: "auto-generated role by the ephemeral access controller",
-		// TODO
-		Policies: []string{},
+		Name:        rt.AppProjectRoleName(ar.Spec.Application.Name, ar.Spec.Application.Namespace),
+		Description: rt.Spec.Description,
+		// TODO replace from template
+		Policies: rt.Spec.Policies,
 		Groups:   groups,
 	}
 	project.Spec.Roles = append(project.Spec.Roles, role)
 }
 
+// updateProjectPolicies will update the given project to match all Policies
+// defined by the given RoleTemplate for the role name specified in the rt.
+// It will also update the description and revoke any JWT tokens that were
+// associated with this specific role. Noop if the given rt is nil.
+func updateProjectPolicies(project *argocd.AppProject, ar *api.AccessRequest, rt *api.RoleTemplate) {
+	if rt == nil {
+		return
+	}
+	roleName := rt.AppProjectRoleName(ar.Spec.Application.Name, ar.Spec.Application.Namespace)
+	for idx, role := range project.Spec.Roles {
+		if role.Name == roleName {
+			project.Spec.Roles[idx].Description = rt.Spec.Description
+			project.Spec.Roles[idx].Policies = rt.Spec.Policies
+			project.Spec.Roles[idx].JWTTokens = nil
```

Comment:
Will `nil` remove the field or it will be considered as "omitempty" ? We may need to set it to `[]`.

Reply:
good catch! It would be considered as "omitempty" indeed. Fixed.